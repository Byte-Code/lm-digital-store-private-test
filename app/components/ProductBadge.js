import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import styled from 'styled-components';

import Image from './Image';

const Wrapper = styled.div`
  height: 605px;
  width: 405px;
  color: #333333;
`;

const Title = styled.div`
  font-size: 16px;
  line-height: 24px;
  text-align: center;
  margin: 10px 0;
`;

const Name = styled.div`
  line-height: 32px;
  font-size: 20px;
  text-align: center;
  font-family: LeroyMerlinSans Bold;
  margin: 10px 0;
  text-transform: uppercase;
`;

const ProductBadge = ({ productInfo }) => {
  const imageID = productInfo.get('mainImage');
  const imageOptions = { width: 405, height: 405 };
  const name = productInfo.get('name');
  const title = 'FACILE DA RIPORRE';

  return (
    <Wrapper>
      <Title>{title}</Title>
      <Image
        imageID={imageID}
        imageOptions={imageOptions}
        alt={name}
      />
      <Name>{name}</Name>
    </Wrapper>
  );
};

ProductBadge.propTypes = {
  productInfo: ImmutablePropTypes.map.isRequired
};

export default ProductBadge;
