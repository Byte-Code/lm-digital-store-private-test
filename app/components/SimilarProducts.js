import React, { Component } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { List } from 'immutable';
import styled from 'styled-components';

import ProductBadge from './ProductBadge';
import SimilarProductsDialog from './SimilarProductsDialog';

const Wrapper = styled.div`
  padding-left: 40px;
  display: flex;
`;

const Header = styled.div`
  width: 90px;
  &>h3 {
    font-size: 16px;
    text-transform: uppercase;
    padding-right: 60px;
  }
`;

const Slider = styled.div`
  display: flex;
  overflow-x: auto;
  &>div {
    margin-right: 20px;
  }
`;

export default class SimilarProducts extends Component {
  static propTypes = {
    similarProducts: ImmutablePropTypes.list,
  }

  static defaultProps = {
    similarProducts: List()
  }

  constructor(props) {
    super(props);
    this.state = {
      dialogOpen: false,
      selectedProduct: ''
    };
  }

  handleOpen = (product) => {
    this.setState({
      selectedProduct: product,
      dialogOpen: true
    });
  }

  handleClose = () => {
    this.setState({ dialogOpen: false });
  }

  renderProducts() {
    const { similarProducts } = this.props;

    return similarProducts.map(p => (
      <ProductBadge
        key={p.get('code')}
        productInfo={p}
        handleClick={() => this.handleOpen(p.get('code'))}
      />
    ));
  }

  render() {
    const { similarProducts } = this.props;
    if (similarProducts.isEmpty()) {
      return null;
    }
    const { dialogOpen, selectedProduct } = this.state;

    return (
      <Wrapper>
        <Header>
          <h3>prodotti simili</h3>
        </Header>
        <Slider>
          {this.renderProducts()}
        </Slider>
        <SimilarProductsDialog
          similarProducts={similarProducts}
          handleClose={this.handleClose}
          isOpen={dialogOpen}
          selectedProduct={selectedProduct}
        />
      </Wrapper>
    );
  }
}
