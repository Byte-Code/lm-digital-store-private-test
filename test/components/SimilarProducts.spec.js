import React from 'react';
import { shallow } from 'enzyme';
import { List, fromJS } from 'immutable';

import SimilarProducts from '../../app/components/SimilarProducts';
import ProductBadge from '../../app/components/ProductBadge';

const emptySimilarProducts = List();
const similarProducts = fromJS([
  { code: 'product0' },
  { code: 'product1' },
  { code: 'product2' }
]);
const title = 'title';

describe('SimilarProducts', () => {
  it('should render null when similarProducts are empty', () => {
    const result = shallow(
      <SimilarProducts
        similarProducts={emptySimilarProducts}
        title={title}
      />
    );
    expect(result).toMatchSnapshot();
  });

  it('should otherwise render a list of similarProducts', () => {
    const result = shallow(
      <SimilarProducts
        similarProducts={similarProducts}
        title={title}
      />
    );
    expect(result.find(ProductBadge)).toHaveLength(3);
    expect(result).toMatchSnapshot();
  });

  it('should have an inital state with dialog closed', () => {
    const result = shallow(
      <SimilarProducts
        similarProducts={similarProducts}
        title={title}
      />
    );
    const expectedState = {
      dialogOpen: false,
      selectedProduct: ''
    };
    expect(result.state()).toEqual(expectedState);
    expect(result).toMatchSnapshot();
  });
});
