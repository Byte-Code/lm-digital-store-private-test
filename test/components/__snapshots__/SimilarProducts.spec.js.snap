// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarProducts should have an inital state with dialog closed 1`] = `
<styled.div>
  <styled.div>
    <h3>
      title
    </h3>
  </styled.div>
  <styled.div>
    <ProductBadge
      handleClick={[Function]}
      productInfo={
        Object {
          "code": "product0",
        }
      }
    />
    <ProductBadge
      handleClick={[Function]}
      productInfo={
        Object {
          "code": "product1",
        }
      }
    />
    <ProductBadge
      handleClick={[Function]}
      productInfo={
        Object {
          "code": "product2",
        }
      }
    />
  </styled.div>
  <SimilarProductsDialog
    handleClose={[Function]}
    isOpen={false}
    selectedProduct=""
    similarProducts={
      Array [
        Object {
          "code": "product0",
        },
        Object {
          "code": "product1",
        },
        Object {
          "code": "product2",
        },
      ]
    }
  />
</styled.div>
`;

exports[`SimilarProducts should otherwise render a list of similarProducts 1`] = `
<styled.div>
  <styled.div>
    <h3>
      title
    </h3>
  </styled.div>
  <styled.div>
    <ProductBadge
      handleClick={[Function]}
      productInfo={
        Object {
          "code": "product0",
        }
      }
    />
    <ProductBadge
      handleClick={[Function]}
      productInfo={
        Object {
          "code": "product1",
        }
      }
    />
    <ProductBadge
      handleClick={[Function]}
      productInfo={
        Object {
          "code": "product2",
        }
      }
    />
  </styled.div>
  <SimilarProductsDialog
    handleClose={[Function]}
    isOpen={false}
    selectedProduct=""
    similarProducts={
      Array [
        Object {
          "code": "product0",
        },
        Object {
          "code": "product1",
        },
        Object {
          "code": "product2",
        },
      ]
    }
  />
</styled.div>
`;

exports[`SimilarProducts should render null when similarProducts are empty 1`] = `null`;
